test/json_formatter_SUITE.erl

```diff
@@ -0,0 +1,392 @@
+                         {format_packet_filter, {fun mongoose_log_filter:format_packet_filter/2, no_state}},
+                         {format_acc_filter, {fun mongoose_log_filter:format_acc_filter/2, no_state}},
+                         {format_c2s_state_filter, {fun mongoose_log_filter:format_c2s_state_filter/2, no_state}},
+                         {format_stacktrace_filter, {fun mongoose_log_filter:format_stacktrace_filter/2, no_state}}
+                        ]},
+    {ok, Backend} = mongoose_logger_running(ConfigFilters),
+    [{logger_primary_config, LoggerConfig}, {logger_backend, Backend} | Config];
+init_per_testcase(format_depth_limited, Config) ->
+    LoggerConfig = logger:get_primary_config(),
+    logger:set_primary_config(level, info),
+    DepthLimit = 2,
+    ConfigFormatter = #{formatter => {mongoose_json_formatter, #{format_depth => DepthLimit}}},
+    {ok, Backend} = mongoose_logger_running(ConfigFormatter),
+    [{logger_primary_config, LoggerConfig}, {logger_backend, Backend}, {format_depth, DepthLimit}| Config];
+init_per_testcase(chars_limited, Config) ->
+    LoggerConfig = logger:get_primary_config(),
+    logger:set_primary_config(level, info),
+    CharsLimit = 30,
+    ConfigFormatter = #{formatter => {mongoose_json_formatter, #{format_chars_limit => CharsLimit}}},
+    {ok, Backend} = mongoose_logger_running(ConfigFormatter),
+    [{logger_primary_config, LoggerConfig}, {logger_backend, Backend}, {chars_limit, CharsLimit}| Config];
+init_per_testcase(json_depth_limited, Config) ->
+    LoggerConfig = logger:get_primary_config(),
+    logger:set_primary_config(level, info),
+    Depth = 5,
+    ConfigFormatter = #{formatter => {mongoose_json_formatter, #{depth => Depth}}},
+    {ok, Backend} = mongoose_logger_running(ConfigFormatter),
+    [{logger_primary_config, LoggerConfig}, {logger_backend, Backend}, {depth, Depth}| Config];
+init_per_testcase(_TC, Config) ->
+    LoggerConfig = logger:get_primary_config(),
+    logger:set_primary_config(level, info),
+    {ok, Backend} = mongoose_logger_running(),
+    [{logger_primary_config, LoggerConfig}, {logger_backend, Backend} | Config].
+
+end_per_testcase(_TC, Config) ->
+    logger:remove_handler(?HID),
+    logger:set_primary_config(?config(logger_primary_config, Config)),
+    ok.
+
+%%
+%% Tests
+%%
+something_is_logged(Config) ->
+    {_, File} = ?config(logger_backend, Config),
+    Before = get_log(File),
+    ?LOG_INFO(#{what => something_is_logged,
+                text => "JSON formatter test. This is an example message.",
+                code => 404, reason => test_execution}),
+    ?assertNotEqual(timeout, get_at_least_n_log_lines(File, length(Before) + 1, timer:seconds(5))).
+
+something_is_formatted(Config) ->
+    {_, File} = ?config(logger_backend, Config),
+    Before = get_log(File),
+
+    ?LOG_INFO(#{in => config, user => #{name => <<"bbby">>, id => 12345}, details => {entropy, too_low},
```

Comment:
`details => {entropy, too_low}` should be printed as `"details": "{entropy, too_low}"`.
Only lists and maps are relatively safe to convert into JSON's lists and objects.


Good:

```
"c2s_sid":"{1599569734211895,<0.4009.0>}"
```

Bad:

```
"c2s_sid":{"1599569734211895":"<0.4009.0>"} // it's not a KV pair, it's Timestamp and Pid
```


We could live with meta as an object for now, we can always flatten it later, if it would cause issues.